src/cargo/util/command_prelude.rs

```diff
@@ -1483,6 +1505,45 @@ fn get_dependencies_from_metadata() -> CargoResult<(Vec<Dependency>, Vec<Depende
     ))
 }
 
+fn get_ws_member_packages() -> CargoResult<Vec<Package>> {
+    let gctx = new_gctx_for_completions()?;
+    let ws = Workspace::new(&find_root_manifest_for_wd(gctx.cwd())?, &gctx)?;
+    let packages = ws.members().map(Clone::clone).collect::<Vec<_>>();
+    Ok(packages)
+}
+
+pub fn get_ws_member_candidates() -> Vec<clap_complete::CompletionCandidate> {
```

Comment:
I feel like this should at least go before `get_direct_dependencies_pkg_name_candidates` to help draw attention to this